test_runner/regress/test_gc_cutoff.py

```diff
@@ -30,10 +31,9 @@ def test_gc_cutoff(neon_env_builder: NeonEnvBuilder, pg_bin: PgBin):
 
     pageserver_http.configure_failpoints(("after-timeline-gc-removed-layers", "exit"))
 
-    for i in range(5):
-        try:
+    for _ in range(5):
+        with pytest.raises(Exception):
             pg_bin.run_capture(["pgbench", "-N", "-c5", "-T100", "-Mprepared", connstr])
-        except Exception:
-            env.pageserver.stop()
-            env.pageserver.start()
-            pageserver_http.configure_failpoints(("after-timeline-gc-removed-layers", "exit"))
+        env.pageserver.stop()
+        env.pageserver.start()
+        pageserver_http.configure_failpoints(("after-timeline-gc-removed-layers", "exit"))
```

Comment:
This changes what this test does. Previously, it would run pgbench until GC runs and hits the failpoint, and then restart the pageserver. Now, it will restart pageserver after 100s, even if the failpoint was not reached. 

I remember I looked at this test before and it took me a while to understand how it works, so a comment would be in order. And it's a bit fragile anyway; if the failpoint is never hit for some reason, the test will pass. Or it would pass, if it didn't run into the general test timeout we have (5 minutes IIRC).

Was there a particular reason for this change in this PR, or was it just unrelated cleanup?

Reply:
> Now, it will restart pageserver after 100s, even if the failpoint was not reached.

It wont. It will fail the test if failpoint was not reached. See https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises

> And it's a bit fragile anyway; if the failpoint is never hit for some reason, the test will pass.

This is exactly why I added this change. I noticed test failure only because of the timeout. Not because of the failpoint not being hit

Reply:
Ah, I missed the change from try-except to pytest.raises. Ok.